File: Mocktests/src/test/java/com/fourcircles/learning/mock/service/HelloServiceTest.java
package com.fourcircles.learning.mock.service;

import com.fourcircles.learn.mock.service.HelloService;
import org.junit.jupiter.api.Test;
import org.mockito.Mock;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.Mockito.when;

public class HelloServiceTest {
    @Mock
    HelloService helloServiceMock;

    @Test
    void test1(){
      when(helloServiceMock.sayHello("world")).thenReturn("Hello World!");
      System.out.println(helloServiceMock.sayHello("world"));

//      assertEquals("Hello World", helloServiceMock.sayHello("world") );
    }
}

File: collectibles/src/test/kotlin/com/circlesllc/collecitibles/service/HelloServiceTest.kt
package com.circlesllc.collecitibles.service


import com.circlesllc.collectibles.service.HelloService
import io.mockk.every
import io.mockk.mockk
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired


class HelloServiceTest() {


    @Test
    fun `testing my first mockk`() {
//        @MockK
       val helloService: HelloService = mockk(relaxed = true)
        every{helloService.sayHello("rajeev")} returns "Hello rajeev"
        println(helloService.sayHello(("rajeev")))
        Assertions.assertEquals("Hello rajeev",helloService.sayHello(("rajeev")))
    }

    @Test
    fun `testing the real function`(){
//        @Autowired
        var helloService:HelloService = HelloService()
        println(helloService.sayHello("Inigo Montoya"))
        Assertions.assertEquals("Hello Mr Inigo Montoya", helloService.sayHello("Inigo Montoya"))

    }

}